storm-server/src/main/java/org/apache/storm/scheduler/blacklist/strategies/DefaultBlacklistStrategy.java

```diff
@@ -72,13 +74,23 @@ public void prepare(Map<String, Object> conf) {
             }
         }
 
+        // update countMap failures for sendAssignments failing
+        for (Map<String, Integer> item : sendAssignmentFailureCount) {
+            for (Map.Entry<String, Integer> entry : item.entrySet()) {
+                String supervisorNode = entry.getKey();
+                int sendAssignmentFailures = entry.getValue() + countMap.getOrDefault(supervisorNode, 0);
+                countMap.put(supervisorNode, sendAssignmentFailures);
+            }
+        }
+
         for (Map.Entry<String, Integer> entry : countMap.entrySet()) {
             String supervisor = entry.getKey();
             int count = entry.getValue();
             if (count >= toleranceCount) {
                 if (!blacklist.containsKey(supervisor)) { // if not in blacklist then add it and set the resume time according to config
                     LOG.debug("Added supervisor {} to blacklist", supervisor);
                     LOG.debug("supervisorsWithFailures : {}", supervisorsWithFailures);
+                    LOG.debug("sendAssignmentFailureCount: {}", sendAssignmentFailureCount);
                     reporter.reportBlacklist(supervisor, supervisorsWithFailures);
```

Comment:
This `reportBlacklist` no longer fits the need since we added sentAssignmentFailure. Maybe need to fix it as a follow-up. Not sure what's the best way to do it